tests/nat/eval/red_teaming_evaluator/fake_chat_models.py

```
@@ -0,0 +1,112 @@
+from langchain_core.outputs import ChatGeneration
+from langchain_core.outputs import ChatResult
+
+
+class FakeJudgeLLM(BaseChatModel):
+    """
+    A deterministic mock LLM judge that evaluates outputs based on pattern matching.
+    Returns scores based on the presence of specific patterns in the output.
+    """
+
+    # Define patterns and their scores as class attributes. Can be overridden in the constructor.
+    patterns: dict[str, float] = {}
```

Ruff (0.14.8)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)